tools/hangar/__snapshots__/invalid.ts.snap

```diff
@@ -981,24 +981,17 @@ error: Expected type to be \\"inflight (key: str, type: BucketEventType): void\\
    |                               ^^^^ Expected type to be \\"inflight (key: str, type: BucketEventType): void\\", but got \\"inflight (request: ApiRequest): ApiResponse\\" instead
 
 
-error: Expected \\"Json\\" elements to be Json values (https://www.json.org/json-en.html), but got \\"Array<unknown>\\" which is not a Json value
-   --> ../../../examples/tests/invalid/inference.w:71:14
-   |
-71 | Json { cool: anotherEmptyArray };
-   |              ^^^^^^^^^^^^^^^^^ Expected \\"Json\\" elements to be Json values (https://www.json.org/json-en.html), but got \\"Array<unknown>\\" which is not a Json value
-
```

Comment:
hmm, why does this no longer fail?

```ts
let anotherEmptyArray = [];
Json { cool: anotherEmptyArray };
```

Reply:
Array<unknown> either means:

1. It's an empty array. That's always valid Json.
2. It's an array with an inference that we figure out during type checking. With this PR we now validate Json legal values after type checking is complete, so we should be safe to not preemptively show this error
3. It's an array with an inference that we never figure out. No reason to show an error here because it would be a cascading error.